src/components/dashboard/components/dictionary/common/DictionaryModal.jsx

```diff
@@ -47,6 +70,9 @@ export class DictionaryModal extends React.Component {
     if (Object.keys(errors).length === 0) {
       this.props
         .submit(this.state.data)
+        .then(() => {
+          this.props.modalhide();
+        })
         .catch((error) => {
           if (error.response) {
```

Comment:
Yes, it should. The modal has to remain open if there is an error after submission because it is a form. Any error emanating from the server-side will be displayed via a toast notification. For the sake of user experience, the modal form has to be open so the user can submit the form again.